xdsl/dialects/builtin.py

```diff
@@ -1012,6 +1012,7 @@ def unpack(
 @irdl_attr_definition
 class ComplexType(
     Generic[ComplexElementT],
+    PackableType[complex | tuple[int, int]],
```

Comment:
I just played with this locally, and it feels like doing this makes the code easier to read and reason about:
```suggestion
    PackableType[tuple[float, float] | tuple[int, int]],
```
Do we get much utility from making it complex?